clifford/test/test_dpga.py

```diff
@@ -0,0 +1,188 @@
+
+import numpy as np
+import pytest
+from clifford.dpga import *
+
+
+class TestBasicDPGA:
+    def test_non_orthogonal_metric(self):
+        for i, a in enumerate(wbasis):
+            for j, b in enumerate(wbasis):
+                res = a|b
+                if i != j and i%4 == j%4:
+                    assert res == (0.5) + 0*w1
+                else:
+                    assert res == 0*w1
+                print(a|b, end=' ')
+            print()
+
+    def test_bivector_identities(self):
+        for wi in wlist:
+            for wj in wlist:
+                assert wi^wj == -wj*wi
+
+        for wis in wslist:
+            for wjs in wslist:
+                assert wis^wjs == -wjs*wis
+
+        for w in wbasis:
+            assert w**2 == 0*w1
+
+        for wi, wis in zip(wlist, wslist):
+            assert wi*wis == 1 - wis*wi
```

Comment:
Would be super nice if you could give an equation number from the paper for these identities